meta/lib/oe/packagefeed.py

```diff
@@ -0,0 +1,120 @@
+#
+# SPDX-License-Identifier: GPL-2.0-only
+#
+
+from abc import ABCMeta, abstractmethod
+import os
+import oe
+
+class Packagefeed(object, metaclass=ABCMeta):
+    """
+    This is an abstract class. Do not instantiate this directly.
+    """
+
+    def __init__(self, d):
+        self.d = d
+        self.deploy_dir = self.d.expand('${DEPLOY_DIR_FEED}/${PN}')
+        self.indexer = None
+
+    @abstractmethod
+    def _create_feed_directory(self):
+        pass
+
+    def create(self):
+        bb.note("###### Generate packagefeed with index #######")
+
+        self._create_feed_directory()
+
+        # call the package indexer create index method
+        self.indexer.write_index()
+
+def get_class_for_type(imgtype):
+    import importlib
+    mod = importlib.import_module('oe.package_manager.' + imgtype + '.packagefeed')
+    return mod.PkgFeed
+
+def create_packagefeed(d):
+    env_bkp = os.environ.copy()
+
+    img_type = d.getVar('IMAGE_PKGTYPE')
+
+    cls = get_class_for_type(img_type)
+    cls(d).create()
+    os.environ.clear()
+    os.environ.update(env_bkp)
```

Comment:
This backs up the environment variables (in case anything changes during the process), instantiates an instance of the proper packagefeed class implementation based on the IMAGE_PKGTYPE, and then calls the create() method to create a packagefeed.